lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -3645,7 +3645,10 @@ static int prLanguageConfig_writeConfigFile(struct VMGlobals* g, int numArgsPush
         config_path = SC_Codecvt::utf8_str_to_path(path);
         gLanguageConfig->writeLibraryConfigYAML(config_path);
     } else {
-        config_path = SC_Filesystem::instance().getDirectory(SC_Filesystem::DirName::UserConfig) / "sclang_conf.yaml";
+        config_path = SC_Codecvt::path_to_utf8_str(gLanguageConfig->getConfigPath());
+        if (config_path.size() == 0)
+            config_path =
+                SC_Filesystem::instance().getDirectory(SC_Filesystem::DirName::UserConfig) / "sclang_conf.yaml";
```

Comment:
`"sclang_conf.yaml"` is harcoded here, while we have it defined [here](https://github.com/supercollider/supercollider/blob/3037bf299bf21618bdf11fce53bf53427e84f13a/lang/LangSource/SC_LanguageConfig.cpp#L44). Would there be a way to use that constant here?

Reply:
sure! probably the easiest thing to do would be add `extern const char* SCLANG_YAML_CONFIG_FILENAME` in the header, and remove the `static` from the cpp file.

Reply:
do you mean a new header file?

Reply:
nah, it can go in the sc language config header file

Reply:
`SCBase.h`?

Reply:
yes

Reply:
Should I move all these or just `SCLANG_YAML_CONFIG_FILENAME`?


```supercollider
static const char* INCLUDE_PATHS = "includePaths";
static const char* EXCLUDE_PATHS = "excludePaths";
static const char* POST_INLINE_WARNINGS = "postInlineWarnings";
static const char* CLASS_LIB_DIR_NAME = "SCClassLibrary";
static const char* SCLANG_YAML_CONFIG_FILENAME = "sclang_conf.yaml";
```

Reply:
just the one, unless you see places where the others could be used. to clarify, you are not moving anything, but rather adding an extern declaration in the header and removing static here.

Reply:
thanks, that's helpful. I think I understand now.